partner/pmp-partner-service/src/main/java/io/mosip/pmp/partner/controller/PartnerServiceController.java

```diff
@@ -325,4 +331,28 @@
 		response.setResponse(partnerService.getPartnerCertificate(requestDto));
 		return response;
     }	
+	
+	@ResponseFilter
+	@PostMapping("/searchPartner")
+	// @PreAuthorize("hasAnyRole('PARTNER','PMS_USER','AUTH_PARTNER','DEVICE_PROVIDER','FTM_PROVIDER','CREDENTIAL_PARTNER','CREDENTIAL_ISSUANCE','CREATE_SHARE','ID_AUTHENTICATION')")
+	public ResponseWrapper<PageResponseDto<PartnerSearchDto>> searchPartner(
+			@RequestBody @Valid RequestWrapper<SearchDto> request) {
+		ResponseWrapper<PageResponseDto<PartnerSearchDto>> responseWrapper = new ResponseWrapper<>();
+
+		responseWrapper.setResponse(partnerService.searchPartner(request.getRequest()));
+
+		return responseWrapper;
+	}
+
+	@ResponseFilter
+	@PostMapping("/searchPartnerType")
+	// @PreAuthorize("hasAnyRole('PARTNER','PMS_USER','AUTH_PARTNER','DEVICE_PROVIDER','FTM_PROVIDER','CREDENTIAL_PARTNER','CREDENTIAL_ISSUANCE','CREATE_SHARE','ID_AUTHENTICATION')")
```

Comment:
1. Can we change api name as /partnerType/search
2. Please enable roles

Reply:
Resolved